### PR TITLE
Add Core CMS and GraphQl Features

### PR DESCRIPTION
**Pull request by Blake-Haydon**
## Description

This main focus of this PR is to initialize netlify CMS and add dynamic content to the site.

Important changes:
- Netlify CMS added (gatsby-plugin-netlify-cms)
- Added initial markdown files for CMS
- Added dynamic linking to the markdown (useStaticQuery and graphql)
- Netlify form added

Smaller changes:
- Button component was made more universal, allowing to be both a link and a button (This was done so that one component could be used for both links and form submission)
- Global CSS cleanup
- Default props fixed and cleaned up

### _Fancy_ dynamic content using the CMS and graphQl
image

<br/>

### Correct breakdown on moblie (with added padding)
image

<br/>

### Shiny new CMS 

image

## Steps to Test (Locally)

Use the instructions in the [readme](https://github.com/monash-human-power/MHP-website#readme)

## Steps to Test (Live)

View the site [here](https://deploy-preview-6--mhp-test.netlify.app/ )
_There is a bug where the Facebook logo will be really large. It has been noted and I'm looking into it_

### CODE REVIEW

File: src/components/button.js [hunk 1]
```diff
@@ -4,37 +4,63 @@ import propTypes from "prop-types";
...
+  &:hover {
+    /* Slightly darker purple */
+    background-color: #37279e;
+  } 
+`;
+
+const LinkButton = styled(Link)`
+  ${universalButtonStyle}
+`;
+const RealButton = styled.button`
```
Comment (hallgchris) [inline src/components/button.js:28]: Would `SubmitButton` be more descriptive?

File: src/components/info_block.js [hunk 2]
```diff
@@ -55,13 +52,11 @@ InfoBlock.propTypes = {
 };
 
 InfoBlock.defaultProps = {
-  heading: "Heading example",
-  description:
-    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
-  buttonText: "Example Button Text",
+  heading: "",
+  description: "",
+  buttonText: "",
```
Comment (hallgchris) [inline src/components/info_block.js:57]: Should we have a default image as well? Or at least handle when one isn't provided

File: src/components/socials.js [hunk 3]
```diff
@@ -28,17 +29,36 @@ const SocialLink = styled(Link)`
...
 
 const Socials = () => {
+  const data = useStaticQuery(graphql`
+    query socialQuery {
+      file(
+        relativePath: { eq: "socials.md" }
+        sourceInstanceName: { eq: "markdown" }
+      ) {
+        childMarkdownRemark {
+          frontmatter {
```
Comment (hallgchris) [inline src/components/socials.js:39]: Do you want to do the weird renaming thing that @khanguslee spoke about yesterday?

File: src/components/socials.js [hunk 4]
```diff
@@ -28,17 +29,36 @@ const SocialLink = styled(Link)`
...
         <FontAwesomeIcon icon={faFacebookSquare} />
       </SocialLink>
 
-      <SocialLink to="https://www.instagram.com/monashhpt/">
+      <SocialLink to={social_link.instagram}>
         <FontAwesomeIcon icon={faInstagramSquare} />
       </SocialLink>
 
-      <SocialLink to="https://www.linkedin.com/company/monashhpt/">
+      <SocialLink to={social_link.linkedIn}>
```
Comment (hallgchris) [inline src/components/socials.js:61]: Would it be useful for Corp to be able to add whatever new links they want here by just specifying a link and a FA icon? (we could tell them what that is)

File: src/pages/index.js [hunk 5]
```diff
@@ -1,69 +1,160 @@
...
+              // Example key would be
+              id={blockData.id}
+              // Flips the order for every second block
+              reverseOrder={index % 2 === 1}
+            />
+          ))}
+        </div>
+
+        {/* Contact form */}
+        <a name="contact" />
```
Comment (hallgchris) [inline src/pages/index.js:94]: Should this be its own component?

File: src/components/button.js (hunk 1 above)
Comment (Blake-Haydon) [reply src/components/button.js:28]: Definitely would be!

File: src/components/socials.js (hunk 3 above)
Comment (Blake-Haydon) [reply src/components/socials.js:39]: I looked into that and I cant find a way to do it in Gatsby. I think you have to traverse the whole JSON object to find the data you want.

File: src/pages/index.js (hunk 5 above)
Comment (Blake-Haydon) [reply src/pages/index.js:94]: Yes! it has been refactored now

**Comment by Blake-Haydon** [comment]
I know that the website isn't finished but even if its not perfect I would really like to push this into the master!

File: src/components/index/contact_form.js [hunk 6]
```diff
@@ -47,14 +47,14 @@ const ContactForm = () => (
...
             required
           />
         </div>
 
         {/* Submit Button */}
-        <div className="row">
-          <div className="col">
+        <div className="row justify-content-center">
+          <div className="col-5">
             <Button type="submit">Submit</Button>
```
Comment (hallgchris) [inline src/components/index/contact_form.js:58]: Silly question but does this form actually submit?

File: src/pages/bike.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+import SEO from "../components/seo";
+import InfoBlock from "../components/info_block";
+
+import SubpageHeading from "../components/subpage_heading";
+import styled from "styled-components";
+
+import MHP_purple_crosshair from "../images/crosshair_purple.svg";
+
+const CrosshairLayout = styled(Layout)`
+  background: repeat center/50px url(${MHP_purple_crosshair});
```
Comment (hallgchris) [inline src/pages/bike.js:15]: This seems to be the only page with the crosshair background. Are we going to have this on all pages?

File: src/pages/bike.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  margin: 0px;
+`;
+
+const StatsInfo = styled.h2`
+  background: var(--MHP-purple);
+  color: var(--MHP-white);
+`;
+
+const WhiteDiv = styled.div`
+  background: var(--MHP-white);
```
Comment (hallgchris) [inline src/pages/bike.js:30]: In figma, all the pages have a black background - are we going for black or white?

**Review by hallgchris — APPROVED** [review]
Just got a few questions but overall I'm happy with this! Can't wait to see the website live!

**Pull request merged**